pypesto/objective/aggregate.py

```diff
@@ -0,0 +1,175 @@
+        init_kwargs = {
+            'x_names': x_names,
+            'options': options
+        }
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in ['fun_accept_sensi_orders', 'res_accept_sensi_orders']:
+            _check_boolean_value_consistent(objectives, attr)
+            init_kwargs[attr] = getattr(objectives[0], attr)
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in [
+            'fun', 'grad', 'hess', 'hessp', 'res', 'sres'
+        ]:
+            if any(
+                getattr(objective, attr) is None
+                for objective in objectives
+            ):
+                init_kwargs[attr] = None
+            elif all(
+                isinstance(getattr(objective, attr), bool)
+                for objective in objectives
+            ):
+                _check_boolean_value_consistent(objectives, attr)
+                init_kwargs[attr] = getattr(objectives[0], attr)
+            elif all(
+                callable(getattr(objective, attr))
+                for objective in objectives
+            ):
+                aggregate_fun = f'aggregate_{attr}'
+                if (
+                        attr == 'fun'
+                        and init_kwargs['fun_accept_sensi_orders']
+                ) or (
+                        attr == 'res'
+                        and init_kwargs['res_accept_sensi_orders']
+                ):
+                    aggregate_fun += '_sensi_orders'
+
+                init_kwargs[attr] = getattr(self, aggregate_fun)
+            else:
+                raise ValueError(f'{attr} has incompatible types across '
+                                 f'instances!')
+
+        super().__init__(**init_kwargs)
+
+    def aggregate_fun_sensi_orders(self, x, sensi_orders):
+        rvals = [
+            objective.fun(x, sensi_orders)
+            for objective in self.objectives
+        ]
+        return {
+            key: sum(rval[key] for rval in rvals)
+            for key in ['fval', 'grad', 'hess']
+            if key in rvals[0]
+        }
+
+    def aggregate_res_sensi_orders(self, x, sensi_orders):
+        # initialize res and sres
+        rval0 = self.objectives[0].res(x, sensi_orders)
+        if 'res' in rval0:
+            res = np.asarray(rval0['res'])
+        else:
+            res = None
+
+        if 'sres' in rval0:
+            sres = np.asarray(rval0['sres'])
+        else:
+            sres = None
+
+        # skip iobj=0 after initialization, stack matrices
+        for iobj in range(1, len(self.objectives)):
+            rval = self.objectives[iobj].res(x, sensi_orders)
+            if res is not None:
+                res = np.hstack([res, np.asarray(rval['res'])])
+            if sres is not None:
+                sres = np.vstack([sres, np.asarray(rval['sres'])])
```

Comment:
a problem might occur when one of the objectives errors, I'm not sure what happens then. maybe just wait and see if anything breaks then.

Reply:
Should be fine, although we currently do not do any early termination ...